e2eTest/src/main/java/com/google/android/ground/e2etest/SurveyRunnerTest.kt

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.ground.e2etest
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.By
+import androidx.test.uiautomator.Direction
+import androidx.test.uiautomator.UiDevice
+import androidx.test.uiautomator.Until
+import com.google.android.ground.R
+import com.google.android.ground.e2etest.TestConfig.GROUND_PACKAGE
+import com.google.android.ground.e2etest.TestConfig.LONG_TIMEOUT
+import com.google.android.ground.e2etest.TestConfig.SHORT_TIMEOUT
+import com.google.android.ground.e2etest.TestConfig.TEST_SURVEY_IDENTIFIER
+import com.google.android.ground.e2etest.TestConfig.TEST_SURVEY_TASKS_ADHOC
+import com.google.android.ground.model.task.Task
+import junit.framework.TestCase.fail
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class SurveyRunnerTest : AutomatorRunner {
+
+  override lateinit var device: UiDevice
+
+  private val selectSurveyTitle = stringResource(R.string.select_survey_title)
+  private val collectData = stringResource(R.string.collect_data)
+  private val newDataCollectionSite = stringResource(R.string.new_site)
+  private val close = stringResource(R.string.close)
+  private val dropPin = stringResource(R.string.drop_pin)
+  private val addPoint = stringResource(R.string.add_point)
+  private val complete = stringResource(R.string.complete_polygon)
+  private val capture = stringResource(R.string.capture)
+  private val camera = stringResource(R.string.camera)
+  private val ok = stringResource(R.string.ok)
+  private val next = stringResource(R.string.next)
+  private val done = stringResource(R.string.done)
+  private val save = stringResource(R.string.save)
```

Comment:
Idea: Since these are mostly (all?) used in calls to `By.text()`, can we add a helper, e.g. `byText()` which take the id and resolve the string resource directly? This way we don't need to maintain this block of assignments.

Reply:
Good call! The `R.string.<id>` is already doing the heavy lifting of standardization. Done.